### PR TITLE
fix some issues in protocol layer

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/rest_client.hpp [hunk 1]
```diff
@@ -2210,7 +2210,7 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
         std::string FileAttributes;
         Nullable<std::string> FileCreationTime;
         Nullable<std::string> FileLastWriteTime;
-        Nullable<DateTime> FileChangeTime;
+        Nullable<std::string> FileChangeTime;
```
Comment (EmmaZhu) [inline sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/rest_client.hpp:2213]: Would this change be a breaking in interface?

**Review by EmmaZhu — APPROVED** [review]

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/rest_client.hpp (hunk 1 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/rest_client.hpp:2213]: No, it's all in detail namespace, and we're still working on this feature.

File: sdk/storage/azure-storage-files-shares/swagger/README.md [hunk 2]
```diff
@@ -219,6 +219,8 @@ directive:
       $.ShareEnabledProtocols["enum"] = ["Smb", "Nfs"];
       $.ShareEnabledProtocols["x-ms-enum"] = {"name": "ShareProtocols", "modelAsString": false};
       $.ShareEnabledProtocols["x-ms-enum"]["values"] = [{"value": "SMB", "name": "Smb"},{"value": "NFS", "name": "Nfs"}];
+      delete $.FileChangeTime.format;
+      $.FileLastWriteTimeMode["x-ms-enum"]["values"] = [{"value": "now", "name": "Now"},{"value": "preserve", "name": "Preserve"}];
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-files-shares/swagger/README.md:223]: Why do we have to modify the swagger in this case?
Is the swagger wrong, and do other languages behave the same way?

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-shares/swagger/README.md:223]: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.Files.Shares/src/autorest.md#formats-arent-required

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-shares/swagger/README.md:223]: Other languages also do this.

**Pull request merged**